include/gzip/decompress.hpp

```diff
@@ -36,41 +36,32 @@ class Decompressor
 
     template <typename OutputType>
     void decompress(OutputType& output,
-                    const char* data,
+                    char const* data,
                     std::size_t size) const
     {
-
-#ifdef DEBUG
-        // Verify if size (long type) input will fit into unsigned int, type used for zlib's avail_in
-        std::uint64_t size_64 = size * 2;
-        if (size_64 > std::numeric_limits<unsigned int>::max())
-        {
-            throw std::runtime_error("size arg is too large to fit into unsigned int type x2");
-        }
-#endif
-        if (size > max_ || (size * 2) > max_)
-        {
-            throw std::runtime_error("size may use more memory than intended when decompressing");
-        }
-
         // https://github.com/kaorimatz/libdeflate-ruby/blob/0e33da96cdaad3162f03ec924b25b2f4f2847538/ext/libdeflate/libdeflate_ext.c#L340
         // https://github.com/ebiggers/libdeflate/commit/5a9d25a8922e2d74618fba96e56db4fe145510f4
         std::size_t actual_size;
-        std::size_t uncompressed_size_guess = size * 4;
-        output.reserve(uncompressed_size_guess);
+        std::size_t uncompressed_size_guess = std::min(size * 4, max_);
```

Comment:
What happens when `size * 4` will not fit within `std::numeric_limits<std::size_t>::max()`?